src/game/common/rts/science.cpp

```diff
@@ -68,3 +68,42 @@ ScienceType ScienceStore::Lookup_Science(const char *name)
     captainslog_relassert(false, 0xDEAD0006, "Failed to find matching ScienceType in Science Store.");
     return SCIENCE_INVALID;
 }
+
+void ScienceStore::Parse_Science_Definition(INI *ini)
+{
+    const auto *name = ini->Get_Next_Token();
+    const auto key = g_theNameKeyGenerator->Name_To_Key(name);
```

Comment:
`const auto &key` ?

Reply:
It's a value type. Faster to pass by value.

Reply:
Ok. It's not obvious in Review what this is due `auto`.

Reply:
And it would be just as non obvious from `const NameKey key`

Reply:
Also to top it off NameKey would have been the wrong type. So just goes to show typing out the type when not required can and will cause mistakes. The correct type is ScienceType which is a sort of alias of NameKey.